hamilflow/models/brownian_motion.py

```diff
@@ -183,17 +185,29 @@ def _trajectory(self, n_new_steps: int, seed: int) -> np.ndarray:
 
         return trajectory
 
-    def __call__(self, n_steps: int, seed: int = 42) -> pd.DataFrame:
-        """Simulate the coordinates of the particle
+    def generate_from(self, n_steps: int, seed: int = 42) -> pd.DataFrame:
+        """generate data from a set of interpretable params for this model
 
         :param n_steps: total number of steps to be simulated, including the inital step.
         :param seed: random generator seed for the stochastic process.
             Use it to reproduce results.
         """
+        time_steps = np.arange(0, n_steps) * self.system.delta_t
+
+        return self(t=time_steps, seed=seed)
+
+    def __call__(self, t: TypeTime, seed: int = 42) -> pd.DataFrame:
+        """Simulate the coordinates of the particle
+
+        :param t: the time sequence to be used to generate data
+        :param seed: random generator seed for the stochastic process.
+            Use it to reproduce results.
+        """
+        n_steps = len(t)
         trajectory = self._trajectory(n_new_steps=n_steps - 1, seed=seed)
 
         df = pd.DataFrame(trajectory, columns=self._axis_names)
```

Comment:
```suggestion
        df = pd.DataFrame(trajectory, columns=self._axis_names, index=t)
```

Reply:
refactored in 4dd093c

I didn't choose t as index as we kept t as a normal column in other classes. Also in many popular time series forecasting packages such as nixtla, time is not index.